core/dbt/parser/unit_tests.py

```diff
@@ -254,12 +259,16 @@ def _load_rows_from_seed(self, ref_str: str) -> List[Dict[str, Any]]:
     def parse(self) -> ParseResult:
         for data in self.get_key_dicts():
             unit_test = self._get_unit_test(data)
-            model_name_split = unit_test.model.split()
             tested_model_node = self._find_tested_model_node(unit_test)
             unit_test_case_unique_id = (
                 f"{NodeType.Unit}.{self.project.project_name}.{unit_test.model}.{unit_test.name}"
             )
-            unit_test_fqn = [self.project.project_name] + model_name_split + [unit_test.name]
+            unit_test_fqn = self._build_fqn(
+                self.project.project_name,
+                self.yaml.path.original_file_path,
+                unit_test.model,
+                unit_test.name,
+            )
```

Comment:
I took this for a spin locally and it worked as expected! 🎩 

```yaml
--- dbt_project.yml
...
unit-tests:
  jaffle_shop:
    unit_testing:
      my_model:
        +tags: project
```

schema.yml in models/unit_testing/schema.yml

```
dbt build -s +my_model,test_type:unit --exclude tag:project
```

=> finds no unit tests. moving the config around at various levels also worked as expected. 

Would be good to add some functional testing around this either in this PR or in a follow-on issue if it balloons the scope too much. We do have tests on the fqn in tests/unit/test_unit_test_parser.py so for the scope of this refactor those are good!